bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -648,6 +655,13 @@ void stop() {
             allocatorExecutor.shutdown();
             LOG.info("Stopped entry logger preallocator.");
         }
+
+        /**
+         * get the preallocation for tests.
+         */
+        protected Future<BufferedLogChannel> getPreallocation(){
```

Comment:
nit: I think `getPreallocationFuture()` is better. You can remove `protected` to make the method package private, because the test and class are in same package.